### ISSUE TITLE
Include conversion loss in recommended price threshold; make charge/discharge efficiency configurable

### ISSUE THREAD

**Issue by woopstar** [woopstar/hsem#478]
## Problem

The `calculate_recommended_threshold()` function currently computes battery depreciation only:

```python
threshold = (purchase_price × capacity_loss_pct) / (2 × cycles × usable_capacity)
```

But **conversion loss** (energy lost to charge/discharge inefficiency) is a real economic cost that the pre-filter profitability guards ignore. This can cause the planner to approve cycles that actually lose money when conversion loss is high.

### 1. Add conversion loss to the threshold formula

Update `calculate_recommended_threshold()` in `utils/misc.py` to optionally include conversion loss:

```python
threshold = depreciation + conversion_loss
```

Where:
- `depreciation = (price × loss_pct) / (2 × cycles × capacity)`
- `conversion_loss = 1 / (charge_eff × discharge_eff) - 1`

### 2. Update default efficiencies to 98%

The Huawei LUNA2000 series has a round-trip efficiency of ~96% under optimal conditions. Change defaults from 95% to 98% for both charge and discharge efficiency.

### 3. Add charge/discharge efficiency to the config/options flow

Currently these are hardcoded in `const.py` and read in `config_reader.py` but have **no UI**. Add slider selectors to a config flow step (e.g. the init step or a dedicated battery step).

### 4. Eliminate double-counting

- **`engine_core._schedule_slots`** — currently computes `rlp = (1 - cd*dd) * 100` separately and passes it as `conversion_loss_pct` to `apply_arbitrage_grid_charge`. This is a price-proportional term for schedule arbitrage, NOT the same as the fixed add-on in the threshold. No double-counting.
- **Cost function** — already uses `charge_loss_fraction` and `discharge_loss_fraction` as money terms. This is scoring, not pre-filtering. No double-counting.
- **MILP energy balance** — models physics correctly with efficiencies. No double-counting.

### Acceptance criteria

- [ ] `calculate_recommended_threshold()` accepts optional efficiency params
- [ ] Defaults changed to 98%
- [ ] Charge/discharge efficiency configurable from UI (config + options flow)
- [ ] Planner pre-filter guards (`_apply_grid_charge`, `_apply_soc_plan`, `apply_opportunistic_charge`) use conversion-inclusive threshold
- [ ] No double-counting of conversion loss
- [ ] All tests pass
- [ ] Lint + quality pass
